### ISSUE TITLE
Auto-type not finding entries in Waterfox

### ISSUE THREAD

**Issue by iGeekFace** [keeweb/keeweb#812]
I'm using macOS 10.13.2 and global auto type works as expected in Safari. However, in Waterfox 56.0.1 (I assume Firefox as well, but haven't tried), it isn't finding entries automatically.

I tried using an addon which adds the URL to the window title as well, and there are still no results. The window appears, but with no related matches - I have to type in the the site name to get the entry.

**Comment by antelle** [comment]
Probably it doesn't implement accessibility API properly, so it's not possible to get the URL. URLs are supported only on Chrome, Safari and those browsers where it's available through accessibility API.

**Comment by iGeekFace** [comment]
Could it be made to match based on URLs included in the title? https://addons.mozilla.org/en-US/firefox/addon/add-url-to-window-title/

**Comment by antelle** [comment]
It tries. What does it print in logs for your browser?

**Comment by iGeekFace** [comment]
Here's the log from one attempt. Note that in other KeePass clients, the entry for this page is found successfully in the same browser.

> [INFO ] 2017-12-19T06:44:24.523Z [sync:Passwords] Stat found same version, not modified
> [INFO ] 2017-12-19T06:44:24.523Z [sync:Passwords] Sync finished no error
> [DEBUG] 2017-12-19T06:45:02.678Z [auto-type] Auto type event null
> [DEBUG] 2017-12-19T06:45:02.684Z [auto-type] Get window title
> [INFO ] 2017-12-19T06:45:03.171Z [launcher] spawn osascript: 0, 487ms
> waterfox
> [INFO ] 2017-12-19T06:45:03.317Z [launcher] spawn osascript: 0, 146ms
> Overview | WHMCS - https://www.whmcs.com/members/clientarea.php
> [DEBUG] 2017-12-19T06:45:03.318Z [auto-type] Window title Overview | WHMCS - https://www.whmcs.com/members/clientarea.php undefined
> [DEBUG] 2017-12-19T06:45:08.817Z [auto-type] Entry selected null
> [DEBUG] 2017-12-19T06:45:08.858Z [auto-type] Hide window

**Comment by antelle** [comment]
I just checked, this way it will try to match only entry title. I.e. if you have `WHMCS` or `www.whmcs.com`as entry title, it should work. Probably we should try to extract urls from titles and match them. However, I'm not sure: this way anyone can just put `https://www.google.com` into website `<title>` and automatically get your password on auto-type.

**Comment by iGeekFace** [comment]
I think it'd be unlikely that you'd be auto filling login details on a malicious site, but to avoid that issue, it could consider it a no-match, but place the matched URLs at the top of the entry selector, so you have to manually choose one. I'm not sure if there's any sort of protection from that built into other KeePass clients that are parsing the URL from the <title> tag.

**Comment by antelle** [comment]
It depends. If it's a phishing website and auto-type also works on it, that's pretty bad, then it's better to put it into advanced settings.
If they're matching urls from title, probably they don't have a protection: there's no way to figure out how title is built.

**Comment by iGeekFace** [comment]
If it's a phishing site, it could just as easily use the website title without the URL included and it'd be matched with the entry, as URLs are not currently part of the check without the accessibility API.

**Comment by antelle** [comment]
Yeah fair point. But yes, we can match urls in titles only if the url is not available through accessibility api, so it will be not that dangerous.
